src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-join-auth-list/smb-join-auth-list.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { Observable, BehaviorSubject, of } from 'rxjs';
+import { switchMap, catchError } from 'rxjs/operators';
+import { SmbService } from '~/app/shared/api/smb.service';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { TableComponent } from '~/app/shared/datatable/table/table.component';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { JoinAuth } from '../smb.model';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+
+@Component({
+  selector: 'cd-smb-join-auth-list',
+  templateUrl: './smb-join-auth-list.component.html',
+  styleUrls: ['./smb-join-auth-list.component.scss']
+})
+export class SmbJoinAuthListComponent extends ListWithDetails implements OnInit {
+  @ViewChild('table', { static: true })
+  table: TableComponent;
+  columns: CdTableColumn[];
+  permission: Permission;
+  tableActions: CdTableAction[];
+  context: CdTableFetchDataContext;
+
+  joinAuth$: Observable<JoinAuth[]>;
+  subject$ = new BehaviorSubject<JoinAuth[]>([]);
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    public actionLabels: ActionLabelsI18n,
+    private smbService: SmbService
+  ) {
+    super();
+    this.permission = this.authStorageService.getPermissions().smb;
+  }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: $localize`ID`,
+        prop: 'auth_id',
+        flexGrow: 2
```

Comment:
i found it a bit weird that it only has 1 column. But I remember you mentioning that you are looking to see more things to expose here? maybe separate PR?

Reply:
It looks a bit wierd indeed... But these resources, [join-auth](https://docs.ceph.com/en/latest/mgr/smb/#join-auth-resource) and [usersgroups](https://docs.ceph.com/en/latest/mgr/smb/#users-and-groups-resource) don't have much else on them other than ID and usernames/passwords 

For now I've only listed the ID since I know it's something we want for sure, but will try to get the feedback next time I demo this to whether to show usernames or not (or anything else) and how to show them, maybe on a dropdown with a details page?

Also @nizamial09 let me know your thoughs if you have any suggestions